frappe/utils/safe_exec.py

```diff
@@ -110,12 +113,16 @@ def get_safe_globals():
 			errprint=frappe.errprint,
 			qb=frappe.qb,
 			get_meta=frappe.get_meta,
+			new_doc=frappe.new_doc,
 			get_doc=frappe.get_doc,
+			get_mapped_doc=get_mapped_doc,
```

Comment:
This is the most sus. But since postprocess/validator callables have to be defined in a restricted context I think it should be fine.